ocaml/xapi/xapi_xenops.ml

```diff
@@ -1714,7 +1717,11 @@ module Xenopsd_metadata = struct
         in
         delete_nolock ~__context id ;
         md
-    )
+      )
+      (* NOTE: This function has a single user and releasing the upper-level
```

Comment:
If it has a single user always, how about making it as `pull_no_lock` so that it could solely rely on its caller to acquire the lock?

Reply:
Ah, I see. The outer lock will be acquired only when "power_state = `Suspended".

Reply:
It won't be called unless `power_state = Suspended`, so I could just make it `pull_no_lock` indeed and leave the reentrant lock refactoring for the future if someone else starts using this function.

Reply:
Yeah, locking/unlocking twice seems a bit strange to me.